terraform/ipfs-mirror/nginx-gateway

```diff
@@ -0,0 +1,52 @@
+server {
+  listen 80;
+  listen [::]:80;
+  
+  server_name ipfs-gateway-__IPFS_MIRROR_INSTANCE__.__DOMAIN_NAME__;
+  return 301 https://$server_name$request_uri;
+}
+
+server {
+  listen 443 ssl http2;
+  listen [::]:443 ssl http2;
+  
+  server_name ipfs-gateway-__IPFS_MIRROR_INSTANCE__.__DOMAIN_NAME__;
+
+  ssl_certificate /etc/letsencrypt/live/ipfs-mirror-__IPFS_MIRROR_INSTANCE__.__DOMAIN_NAME__/fullchain.pem;
+  ssl_certificate_key /etc/letsencrypt/live/ipfs-mirror-__IPFS_MIRROR_INSTANCE__.__DOMAIN_NAME__/privkey.pem;
+  ssl_dhparam /etc/ssl/certs/dhparam.pem;
+
+  # from https://cipherli.st/
+  # and https://raymii.org/s/tutorials/Strong_SSL_Security_On_nginx.html
+  ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+  ssl_prefer_server_ciphers on;
+  ssl_ciphers "EECDH+AESGCM:EDH+AESGCM:AES256+EECDH:AES256+EDH";
+  ssl_ecdh_curve secp384r1;
+  ssl_session_cache shared:SSL:10m;
+  ssl_session_tickets off;
+  ssl_stapling on;
+  ssl_stapling_verify on;
+  resolver 8.8.8.8 8.8.4.4 valid=300s;
+  resolver_timeout 5s;
+  # Disable preloading HSTS for now.  You can use the commented out header line that includes
+  # the "preload" directive if you understand the implications.
+  #add_header Strict-Transport-Security "max-age=63072000; includeSubdomains; preload";
+  add_header Strict-Transport-Security "max-age=63072000; includeSubdomains";
+  add_header X-Frame-Options DENY;
```

Comment:
I think we should remove the `add_header X-Frame-Options DENY;` because having that header means we cannot iframe the player on other pages if needed to. 

Maybe same goes for ipfs-server too.